src/openqaoa-core/algorithms/qaoa/qaoa_workflow.py

```diff
@@ -13,6 +15,97 @@
 from ...optimizers.qaoa_optimizer import get_optimizer
 
 
+def circuit_routing(
+    device: DeviceBase,
+    problem_to_solve: List[List[int]],
+    initial_mapping: Optional[List[int]] = None,
+    routing_algo: str = "greedy",
+) -> tuple:
+    """Post the HTTP request to the routing API and retrieve routed
+    circuit
+
+    Parameters
+    ----------
+    device: DeviceBase
+        The device for which to route the circuit
+    cost_block: List[GateMap]
+        The QUBO problem to be run on QPU
+    initial_mapping: List[int]
+        The initial selection of qubits, if provided by the user.
+        Defaults to the first n-qubits on the device if `None`
+    routing_algo: str
+        The algorithm to generate an optimal SWAP network
+
+    Returns
+    -------
+    tuple[List[List],List[bool],List[int]]
+        The first list specifies the qubit indices for operations,
+        the second list differentiates between Ising and SWAP gates,
+        and the final list returns the final qubit mapping.
+    """
+    device_connecivity = device.connectivity()
+
+    # params for MCTS
+    params = (
```

Comment:
This should be an argument of the function

Reply:
removing this function